checkstyle-tester/README.md

```diff
@@ -224,6 +224,22 @@ Sevntu's current version must be referenced at https://github.com/checkstyle/con
 
 Finally, just run checkstyle-tester as described above.
 
+## ANTLR Grammar Change Testing Procedure
+
+#### Modified Grammar Check Regression Report
+
+After modifying Checkstyle's ANTLR grammar, it is neccessary to compare how Checkstyle's checks are affected by your changes.  This is done by comparing the check violation output of your modified branch of Checkstyle against Checkstyle's master branch. In order to do this, you must separate the modules and settings in the [`checkstyle_checks.xml`](https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml) configuration file, found in the checkstyle repository, into separate configuration files (this helps avoid OOM).  The best way to separate the modules is to determine which typically produce the most violations, such as Indentation and WhiteSpace Checks, and put them each into their own configuration file, then separating the rest into files with roughly ten modules each. You may modify all the modules that depend on external files to use default settings.  **For each of the configuration files, you should use the [`my_check.xml`](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/my_check.xml) file as a base**, and add the modules from `checkstyle_checks.xml` to it. Then `diff.groovy` should be run on all projects in `projects-to-test-on.properties`, using the `diff.groovy` script once for each configuration file.  [See instructions above](https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#basic-difference-report) for "Basic Difference Report".
+
+*Note: if you are still experienceing maven OOM, see https://cwiki.apache.org/confluence/display/MAVEN/OutOfMemoryError*
+
+#### Abstract Syntax Tree Regression Report
+
+When making changes to Checkstyle's grammar, it is important to verify that there is no regression of the AST.  This is accomplished by using the [`launch_diff_antlr.sh`](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff_antlr.sh) script, and selecting (uncommenting) **only** the checkstyle project from [`projects-to-test-on.properties`](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/projects-to-test-on.properties).  To use this script, first you must modify the [`launch_diff_variables.sh`](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff_variables.sh) file to reflect the location of the variables that  [`launch_diff_antlr.sh`](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff_antlr.sh) uses to produce the report. Please note that `PULL_REMOTE` will need to be set to the name of the remote repository where your branch resides. If your branch is found at `origin/name-of-your-branch-here`, you will set
```

Comment:
I still don't see the reason for this separation. the antlr diff is still a diff on the grammar. If you just want to separate the 2 different types of runs, then I would start with the intro under the main header, `ANTLR Grammar Change Testing Procedure` and then go into the specifics of each run in the sub-header.

Reply:
Fixed, I think.  Let me know if this is what you had in mind.